<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Using ParmEd with MDAnalysis and OpenMM to simulate a selection of atoms &mdash; MDAnalysis User Guide  documentation</title>
      <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/css/msmb.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/plot_directive.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/nbsphinx-code-cells.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/custom.css" type="text/css" />
    <link rel="shortcut icon" href="../../_static/mdanalysis-logo.ico"/>
    <link rel="canonical" href="https://userguide.mdanalysis.org/examples/other/parmed_sim.html" />
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../_static/documentation_options.js?v=5929fcd5"></script>
        <script src="../../_static/doctools.js?v=888ff710"></script>
        <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
        <script defer="defer" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script src="../../_static/js/versions.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within MDAnalysis User Guide  documentation"
          href="../../_static/opensearch.xml"/>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Alignments and RMS fitting" href="../analysis/alignment_and_rms/README.html" />
    <link rel="prev" title="Other" href="README.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >

          
          
          <a href="../../index.html">
            
              <img src="../../_static/user_guide.png" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                2.6.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting started</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../quickstart.html">Quick start guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../faq.html">Frequently asked questions</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../README.html">Examples</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../quickstart.html">Quick start guide</a></li>
<li class="toctree-l2"><a class="reference internal" href="../constructing_universe.html">Constructing, modifying, and adding to a Universe</a></li>
<li class="toctree-l2"><a class="reference internal" href="../transformations/README.html">Transformations</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="README.html">Other</a><ul class="current">
<li class="toctree-l3 current"><a class="current reference internal" href="#">Using ParmEd with MDAnalysis and OpenMM to simulate a selection of atoms</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#Loading-files:-the-difference-between-ParmEd-and-MDAnalysis">Loading files: the difference between ParmEd and MDAnalysis</a></li>
<li class="toctree-l4"><a class="reference internal" href="#Using-MDAnalysis-to-select-atoms">Using MDAnalysis to select atoms</a></li>
<li class="toctree-l4"><a class="reference internal" href="#Using-ParmEd-and-OpenMM-to-create-a-simulation-system">Using ParmEd and OpenMM to create a simulation system</a></li>
<li class="toctree-l4"><a class="reference internal" href="#References">References</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../analysis/alignment_and_rms/README.html">Alignments and RMS fitting</a></li>
<li class="toctree-l2"><a class="reference internal" href="../analysis/distances_and_contacts/README.html">Distances and contacts</a></li>
<li class="toctree-l2"><a class="reference internal" href="../analysis/trajectory_similarity/README.html">Trajectory similarity</a></li>
<li class="toctree-l2"><a class="reference internal" href="../analysis/structure/README.html">Structure</a></li>
<li class="toctree-l2"><a class="reference internal" href="../analysis/reduced_dimensions/README.html">Dimension reduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../analysis/polymers_and_membranes/README.html">Polymers and membranes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../analysis/volumetric/README.html">Volumetric analyses</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference external" href="https://docs.mdanalysis.org/stable/index.html">API docs</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Data structures</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../universe.html">Universe</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../atomgroup.html">AtomGroup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../groups_of_atoms.html">Groups of atoms</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../selections.html">Atom selection language</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../topology_system.html">The topology system</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Trajectories</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../trajectories/trajectories.html">Trajectories</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../trajectories/slicing_trajectories.html">Slicing trajectories</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../trajectories/transformations.html">On-the-fly transformations</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../units.html">Units and constants</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Input/output</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../reading_and_writing.html">Reading and writing files</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../formats/index.html">Format overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../formats/guessing.html">Guessing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../formats/auxiliary.html">Auxiliary files</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../formats/selection_exporters.html">Selection exporters</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../formats/format_reference.html">Format reference</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../analysis/README.html">Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analysis/README.html#alignments-and-rms-fitting">Alignments and RMS fitting</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analysis/README.html#distances-and-contacts">Distances and contacts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analysis/README.html#trajectory-similarity">Trajectory similarity</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analysis/README.html#structure">Structure</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analysis/README.html#volumetric-analyses">Volumetric analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analysis/README.html#dimension-reduction">Dimension reduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analysis/README.html#polymers-and-membranes">Polymers and membranes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analysis/README.html#hydrogen-bond-analysis">Hydrogen Bond Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analysis/custom_trajectory_analysis.html">Writing your own trajectory analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analysis/custom_parallel_analysis.html">Parallelizing analysis</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Advanced</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../standard_selections.html">Standard residues in MDAnalysis selections</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../advanced_topology.html">Advanced topology concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../datasets.html">Example data</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Contributing</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../contributing.html">Contributing to MDAnalysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contributing_code.html">Contributing to the main codebase</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contributing_docs.html">Contributing to the user guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../preparing_releases_and_hotfixes.html">Preparing a release</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../module_imports.html">Module imports in MDAnalysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../testing.html">Tests in MDAnalysis</a></li>
<li class="toctree-l1"><a class="reference external" href="https://docs.mdanalysis.org/dev/index.html">Development docs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../references.html">References</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu"  style="background: white" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">MDAnalysis User Guide</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../README.html">Examples</a></li>
          <li class="breadcrumb-item"><a href="README.html">Other</a></li>
      <li class="breadcrumb-item active">Using ParmEd with MDAnalysis and OpenMM to simulate a selection of atoms</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/examples/other/parmed_sim.ipynb.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <script src='http://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js'></script>
<script>require=requirejs;</script><section id="Using-ParmEd-with-MDAnalysis-and-OpenMM-to-simulate-a-selection-of-atoms">
<h1>Using ParmEd with MDAnalysis and OpenMM to simulate a selection of atoms<a class="headerlink" href="#Using-ParmEd-with-MDAnalysis-and-OpenMM-to-simulate-a-selection-of-atoms" title="Link to this heading"></a></h1>
<p>Here we use MDAnalysis to convert a ParmEd structure to an MDAnalysis Universe, select a subset of atoms, and convert it back to ParmEd to simulate with OpenMM.</p>
<p><strong>Last updated:</strong> December 2022 with MDAnalysis 2.4.0-dev0</p>
<p><strong>Last updated:</strong> December 2022</p>
<p><strong>Minimum version of MDAnalysis:</strong> 1.0.0</p>
<p><strong>Packages required:</strong></p>
<ul class="simple">
<li><p>MDAnalysis <a class="reference internal" href="#References"><span class="std std-ref">[1, 2]</span></a></p></li>
<li><p>MDAnalysisTests</p></li>
<li><p><a class="reference external" href="http://parmed.github.io/ParmEd/html/index.html">ParmEd</a></p></li>
<li><p><a class="reference external" href="http://openmm.org">OpenMM</a> <a class="reference internal" href="#References"><span class="std std-ref">[3]</span></a></p></li>
</ul>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">parmed</span> <span class="k">as</span> <span class="nn">pmd</span>
<span class="kn">import</span> <span class="nn">MDAnalysis</span> <span class="k">as</span> <span class="nn">mda</span>
<span class="kn">from</span> <span class="nn">MDAnalysis.tests.datafiles</span> <span class="kn">import</span> <span class="n">PRM7_ala2</span><span class="p">,</span> <span class="n">RST7_ala2</span>

<span class="kn">import</span> <span class="nn">warnings</span>
<span class="c1"># suppress some MDAnalysis warnings when writing PDB files</span>
<span class="n">warnings</span><span class="o">.</span><span class="n">filterwarnings</span><span class="p">(</span><span class="s1">&#39;ignore&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<section id="Loading-files:-the-difference-between-ParmEd-and-MDAnalysis">
<h2>Loading files: the difference between ParmEd and MDAnalysis<a class="headerlink" href="#Loading-files:-the-difference-between-ParmEd-and-MDAnalysis" title="Link to this heading"></a></h2>
<p>Both ParmEd and MDAnalysis read a number of file formats. However, while MDAnalysis is typically used to analyse simulations, ParmEd is often used to set them up. This requires ParmEd to read topology parameter information that MDAnalysis typically ignores, such as the equilibrium length and force constants of bonds in the system. For example, the ParmEd structure below.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pprm</span> <span class="o">=</span> <span class="n">pmd</span><span class="o">.</span><span class="n">load_file</span><span class="p">(</span><span class="n">PRM7_ala2</span><span class="p">,</span> <span class="n">RST7_ala2</span><span class="p">)</span>
<span class="n">pprm</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;AmberParm 3026 atoms; 1003 residues; 3025 bonds; PBC (orthogonal); parameterized&gt;
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pprm</span><span class="o">.</span><span class="n">bonds</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;Bond &lt;Atom C [10]; In ALA 0&gt;--&lt;Atom O [11]; In ALA 0&gt;; type=&lt;BondType; k=570.000, req=1.229&gt;&gt;
</pre></div></div>
</div>
<p>When MDAnalysis reads these files in, it does not include that information.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mprm</span> <span class="o">=</span> <span class="n">mda</span><span class="o">.</span><span class="n">Universe</span><span class="p">(</span><span class="n">PRM7_ala2</span><span class="p">,</span> <span class="n">RST7_ala2</span><span class="p">,</span> <span class="nb">format</span><span class="o">=</span><span class="s1">&#39;RESTRT&#39;</span><span class="p">)</span>
<span class="n">mprm</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;Universe with 3026 atoms&gt;
</pre></div></div>
</div>
<p>The bond type simply shows the atom types involved in the connection.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mprm</span><span class="o">.</span><span class="n">atoms</span><span class="o">.</span><span class="n">bonds</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">type</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
(&#39;N3&#39;, &#39;H&#39;)
</pre></div></div>
</div>
<p>If you then convert this Universe to ParmEd, you can see that the resulting Structure is not <code class="docutils literal notranslate"><span class="pre">parametrized</span></code>.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mprm_converted</span> <span class="o">=</span> <span class="n">mprm</span><span class="o">.</span><span class="n">atoms</span><span class="o">.</span><span class="n">convert_to</span><span class="p">(</span><span class="s1">&#39;PARMED&#39;</span><span class="p">)</span>
<span class="n">mprm_converted</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;Structure 3026 atoms; 1003 residues; 3025 bonds; parameterized&gt;
</pre></div></div>
</div>
<p>While the bonds are present, there is no <code class="docutils literal notranslate"><span class="pre">type</span></code> information associated.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mprm_converted</span><span class="o">.</span><span class="n">bonds</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;Bond &lt;Atom N [0]; In ALA 0&gt;--&lt;Atom H1 [1]; In ALA 0&gt;; type=None&gt;
</pre></div></div>
</div>
<p>Therefore, if you wish to use ParmEd functionality that requires parametrization on a MDAnalysis Universe, you need to create that Universe <em>from</em> a ParmEd structure in order to convert it <em>back to</em> something useable in ParmEd.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mprm_from_parmed</span> <span class="o">=</span> <span class="n">mda</span><span class="o">.</span><span class="n">Universe</span><span class="p">(</span><span class="n">pprm</span><span class="p">)</span>
<span class="n">mprm_from_parmed</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;Universe with 3026 atoms&gt;
</pre></div></div>
</div>
<p>Now the bond type is actually a ParmEd Bond object.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mprm_from_parmed</span><span class="o">.</span><span class="n">bonds</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">type</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;Bond &lt;Atom N [0]; In ALA 0&gt;--&lt;Atom H1 [1]; In ALA 0&gt;; type=&lt;BondType; k=434.000, req=1.010&gt;&gt;
</pre></div></div>
</div>
</section>
<section id="Using-MDAnalysis-to-select-atoms">
<h2>Using MDAnalysis to select atoms<a class="headerlink" href="#Using-MDAnalysis-to-select-atoms" title="Link to this heading"></a></h2>
<p>One reason we might want to convert a ParmEd structure into MDAnalysis is to use its sophisticated <a class="reference external" href="https://www.mdanalysis.org/UserGuide/selections.html">atom selection syntax</a>. While ParmEd has its <a class="reference external" href="https://parmed.github.io/ParmEd/html/structure.html#structure-manipulation-slicing-combining-replicating-and-splitting">own ways to select atoms</a>, MDAnalysis allows you to select atoms based on geometric distance.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">water</span> <span class="o">=</span> <span class="n">mprm_from_parmed</span><span class="o">.</span><span class="n">select_atoms</span><span class="p">(</span><span class="s1">&#39;around 5 protein&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">residues</span><span class="o">.</span><span class="n">atoms</span>
<span class="n">protein_shell</span> <span class="o">=</span> <span class="n">mprm_from_parmed</span><span class="o">.</span><span class="n">select_atoms</span><span class="p">(</span><span class="s1">&#39;protein&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="n">water</span>
<span class="n">prm_protein_shell</span> <span class="o">=</span> <span class="n">protein_shell</span><span class="o">.</span><span class="n">convert_to</span><span class="p">(</span><span class="s1">&#39;PARMED&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">prm_protein_shell</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;Structure 155 atoms; 46 residues; 154 bonds; PBC (orthogonal); parameterized&gt;
</pre></div></div>
</div>
</section>
<section id="Using-ParmEd-and-OpenMM-to-create-a-simulation-system">
<h2>Using ParmEd and OpenMM to create a simulation system<a class="headerlink" href="#Using-ParmEd-and-OpenMM-to-create-a-simulation-system" title="Link to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">openmm</span> <span class="k">as</span> <span class="nn">mm</span>
<span class="kn">import</span> <span class="nn">openmm.app</span> <span class="k">as</span> <span class="nn">app</span>
<span class="kn">from</span> <span class="nn">parmed</span> <span class="kn">import</span> <span class="n">unit</span> <span class="k">as</span> <span class="n">u</span>
<span class="kn">from</span> <span class="nn">parmed.openmm</span> <span class="kn">import</span> <span class="n">StateDataReporter</span><span class="p">,</span> <span class="n">MdcrdReporter</span>
</pre></div>
</div>
</div>
<p>You can create an OpenMM simulation system directly from a ParmEd structure, providing that it is parametrized.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">system</span> <span class="o">=</span> <span class="n">prm_protein_shell</span><span class="o">.</span><span class="n">createSystem</span><span class="p">(</span><span class="n">nonbondedMethod</span><span class="o">=</span><span class="n">app</span><span class="o">.</span><span class="n">NoCutoff</span><span class="p">,</span>
                                        <span class="n">constraints</span><span class="o">=</span><span class="n">app</span><span class="o">.</span><span class="n">HBonds</span><span class="p">,</span>
                                        <span class="n">implicitSolvent</span><span class="o">=</span><span class="n">app</span><span class="o">.</span><span class="n">GBn2</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>Here we set the integrator to do Langevin dynamics.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">integrator</span> <span class="o">=</span> <span class="n">mm</span><span class="o">.</span><span class="n">LangevinIntegrator</span><span class="p">(</span>
                        <span class="mi">300</span><span class="o">*</span><span class="n">u</span><span class="o">.</span><span class="n">kelvin</span><span class="p">,</span>       <span class="c1"># Temperature of heat bath</span>
                        <span class="mf">1.0</span><span class="o">/</span><span class="n">u</span><span class="o">.</span><span class="n">picoseconds</span><span class="p">,</span>  <span class="c1"># Friction coefficient</span>
                        <span class="mf">2.0</span><span class="o">*</span><span class="n">u</span><span class="o">.</span><span class="n">femtoseconds</span><span class="p">,</span> <span class="c1"># Time step</span>
                        <span class="p">)</span>
</pre></div>
</div>
</div>
<p>We create the Simulation object and set particle positions.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sim</span> <span class="o">=</span> <span class="n">app</span><span class="o">.</span><span class="n">Simulation</span><span class="p">(</span><span class="n">prm_protein_shell</span><span class="o">.</span><span class="n">topology</span><span class="p">,</span> <span class="n">system</span><span class="p">,</span> <span class="n">integrator</span><span class="p">)</span>
<span class="n">sim</span><span class="o">.</span><span class="n">context</span><span class="o">.</span><span class="n">setPositions</span><span class="p">(</span><span class="n">prm_protein_shell</span><span class="o">.</span><span class="n">positions</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>We now minimise the energy.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sim</span><span class="o">.</span><span class="n">minimizeEnergy</span><span class="p">(</span><span class="n">maxIterations</span><span class="o">=</span><span class="mi">500</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>The reporter below reports energies and coordinates every 100 steps to stdout, but every 10 steps to the <code class="docutils literal notranslate"><span class="pre">ala2_shell.nc</span></code> file.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sim</span><span class="o">.</span><span class="n">reporters</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
        <span class="n">StateDataReporter</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">stdout</span><span class="p">,</span> <span class="mi">100</span><span class="p">,</span> <span class="n">step</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">potentialEnergy</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                          <span class="n">kineticEnergy</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">temperature</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">volume</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                          <span class="n">density</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="p">)</span>
<span class="n">sim</span><span class="o">.</span><span class="n">reporters</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">MdcrdReporter</span><span class="p">(</span><span class="s1">&#39;ala2_shell.trj&#39;</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="n">crds</span><span class="o">=</span><span class="kc">True</span><span class="p">))</span>
</pre></div>
</div>
</div>
<p>We can run dynamics for 500 steps (1 picosecond).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sim</span><span class="o">.</span><span class="n">step</span><span class="p">(</span><span class="mi">500</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
#&#34;Step&#34;,&#34;Time (ps)&#34;,&#34;Potential Energy (kilocalorie/mole)&#34;,&#34;Kinetic Energy (kilocalorie/mole)&#34;,&#34;Total Energy (kilocalorie/mole)&#34;,&#34;Temperature (K)&#34;,&#34;Box Volume (angstrom**3)&#34;,&#34;Density (gram/(item*milliliter))&#34;
100,0.20000000000000015,-623.6779995219885,20.140631869613383,-603.5373676523751,63.74314071570579,45325.8064191062,0.034909350700361955
200,0.4000000000000003,-614.1849904397706,38.40737137186695,-575.7776190679035,121.55559436896063,45325.8064191062,0.034909350700361955
300,0.6000000000000004,-606.5526783580306,48.61919832973248,-557.933480028298,153.87503334950912,45325.8064191062,0.034909350700361955
400,0.8000000000000006,-600.0374380078872,57.988937528818184,-542.0485004790689,183.52934648642113,45325.8064191062,0.034909350700361955
500,1.0000000000000007,-603.2854886173518,79.46589388029852,-523.8195947370533,251.50182419815255,45325.8064191062,0.034909350700361955
</pre></div></div>
</div>
<p>If we write a topology file out from our former <code class="docutils literal notranslate"><span class="pre">protein_shell</span></code> atomgroup, we can load the trajectory in for further analysis.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">protein_shell</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;ala2_shell.pdb&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">u</span> <span class="o">=</span> <span class="n">mda</span><span class="o">.</span><span class="n">Universe</span><span class="p">(</span><span class="s1">&#39;ala2_shell.pdb&#39;</span><span class="p">,</span> <span class="s1">&#39;ala2_shell.trj&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">u</span><span class="o">.</span><span class="n">trajectory</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;TRJReader ala2_shell.trj with 50 frames of 155 atoms&gt;
</pre></div></div>
</div>
</section>
<section id="References">
<h2>References<a class="headerlink" href="#References" title="Link to this heading"></a></h2>
<p>[1] R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler, D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein. <a class="reference external" href="http://conference.scipy.org/proceedings/scipy2016/oliver_beckstein.html">MDAnalysis: A Python package for the rapid analysis of molecular dynamics simulations</a>. In S. Benthall and S. Rostrup, editors, <em>Proceedings of the 15th Python in Science Conference</em>, pages 98-105, Austin, TX, 2016. SciPy, doi:
<a class="reference external" href="https://doi.org/10.25080/majora-629e541a-00e">10.25080/majora-629e541a-00e</a>.</p>
<p>[2] N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein. MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations. <em>J. Comput. Chem</em>. 32 (2011), 2319-2327, <a class="reference external" href="https://dx.doi.org/10.1002/jcc.21787">doi:10.1002/jcc.21787</a>. PMCID:<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3144279/">PMC3144279</a></p>
<p>[3] Peter Eastman, Jason Swails, John D. Chodera, Robert T. McGibbon, Yutong Zhao, Kyle A. Beauchamp, Lee-Ping Wang, Andrew C. Simmonett, Matthew P. Harrigan, Chaya D. Stern, Rafal P. Wiewiora, Bernard R. Brooks, Vijay S. Pande. OpenMM 7: Rapid Development of High Performance Algorithms for Molecular Dynamics. <em>PLoS Comput. Biol.</em> 13:e1005659, 2017.</p>
<p>[4] Hai Nguyen, David A Case, Alexander S Rose. NGLview - Interactive molecular graphics for Jupyter notebooks. <em>Bioinformatics</em>. 34 (2018), 1241–1242, <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btx789">doi:10.1093/bioinformatics/btx789</a></p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="README.html" class="btn btn-neutral float-left" title="Other" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../analysis/alignment_and_rms/README.html" class="btn btn-neutral float-right" title="Alignments and RMS fitting" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2019-2023, Lily Wang, Irfan Alibay, Patricio Barletta, Cédric Bouysset, Jan Domanski, Bjarne Feddersen, Florent Langenfeld, Hugo MacDermott-Opeskin, Rocco Meli, Paul Smith, Mieczyslaw Torchala, Mark Verma, Yuxuan Zhuang, Richard J. Gowers, and Oliver Beckstein..</p>
  </div>

  
 
<div class="footer"><p>Please see
    our <a href="https://www.mdanalysis.org/pages/privacy/">Privacy Policy</a>
    to learn how <a href="https://www.mdanalysis.org">MDAnalysis</a> collects data.</p>
    <script data-goatcounter="https://mdanalysis.goatcounter.com/count"
        async src="//gc.zgo.at/count.js"></script>
</div>


</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
    var versions_json_url = 'https://userguide.mdanalysis.org/versions.json'
</script>

<div class="rst-versions" data-toggle="rst-versions" role="note"
     aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      <span class="fa fa-book"></span>
        
      <span class="fa fa-caret-down"></span>
    </span>

    <div class="rst-other-versions">
        <dl id="versionselector">
            <dt>Other Versions</dt>
        </dl>

    </div>
</div><script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>