


  
    
  




<head>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.3/css/all.min.css" integrity="sha512-iBBXm8fW90+nuLcSKlbmrPcLa0OT92xO1BIsZ+ywDWZCvqsWgccV3gFoRBv0z+8dLJgyAHIhR35VZc2oM/gI1w==" crossorigin="anonymous" referrerpolicy="no-referrer" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.9.1/css/academicons.min.css" integrity="sha512-b1ASx0WHgVFL5ZQhTgiPWX+68KjS38Jk87jg7pe+qC7q9YkEtFq0z7xCglv7qGIs/68d3mAp+StfC8WKC5SSAg==" crossorigin="anonymous" referrerpolicy="no-referrer" />
  <link rel="shortcut icon" href="_static/logo/mda_favicon.ico">
</head>

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>References &mdash; MDAnalysis User Guide  documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/site.css" type="text/css" />
      <link rel="stylesheet" href="_static/plot_directive.css" type="text/css" />
    <link rel="canonical" href="https://userguide.mdanalysis.org/references.html" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script src="_static/js/versions.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within MDAnalysis User Guide  documentation"
          href="_static/opensearch.xml"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Tests in MDAnalysis" href="testing.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >




  




<a href="index.html">
  
    <img src="_static/user_guide.png" class="logo" alt="Logo"/>
</a>


  
  
  
    <div class="version">
      2.7.0-dev0
    </div>
  



<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

        </div>
<div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
    
    <!-- <p class="caption" role="heading"></p> -->
    <ul>
        
        <li class="toctree-l1"><a class="reference internal" href="https://mdanalysis.org">MDAnalysis</a></li>
        
        <li class="toctree-l1"><a class="reference internal" href="https://docs.mdanalysis.org/stable">API docs (stable)</a></li>
        
        <li class="toctree-l1"><a class="reference internal" href="https://docs.mdanalysis.org/dev">API docs (development)</a></li>
        
        <li class="toctree-l1"><a class="reference internal" href="https://mdakits.mdanalysis.org/">MDAKits</a></li>
        
    </ul>
    
        <p class="caption" role="heading"><span class="caption-text">Getting started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/quickstart.html">Quick start guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="faq.html">Frequently asked questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/README.html">Examples</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Data structures</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="universe.html">Universe</a></li>
<li class="toctree-l1"><a class="reference internal" href="atomgroup.html">AtomGroup</a></li>
<li class="toctree-l1"><a class="reference internal" href="groups_of_atoms.html">Groups of atoms</a></li>
<li class="toctree-l1"><a class="reference internal" href="selections.html">Atom selection language</a></li>
<li class="toctree-l1"><a class="reference internal" href="topology_system.html">The topology system</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Trajectories</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="trajectories/trajectories.html">Trajectories</a></li>
<li class="toctree-l1"><a class="reference internal" href="trajectories/slicing_trajectories.html">Slicing trajectories</a></li>
<li class="toctree-l1"><a class="reference internal" href="trajectories/transformations.html">On-the-fly transformations</a></li>
<li class="toctree-l1"><a class="reference internal" href="units.html">Units and constants</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Input/output</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="reading_and_writing.html">Reading and writing files</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats/index.html">Format overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats/guessing.html">Guessing</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats/auxiliary.html">Auxiliary files</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats/selection_exporters.html">Selection exporters</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats/format_reference.html">Format reference</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html">Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#alignments-and-rms-fitting">Alignments and RMS fitting</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#distances-and-contacts">Distances and contacts</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#trajectory-similarity">Trajectory similarity</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#structure">Structure</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#volumetric-analyses">Volumetric analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#dimension-reduction">Dimension reduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#polymers-and-membranes">Polymers and membranes</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#hydrogen-bond-analysis">Hydrogen Bond Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/custom_trajectory_analysis.html">Writing your own trajectory analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/custom_parallel_analysis.html">Parallelizing analysis</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Advanced</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="standard_selections.html">Standard residues in MDAnalysis selections</a></li>
<li class="toctree-l1"><a class="reference internal" href="advanced_topology.html">Advanced topology concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="datasets.html">Example data</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Contributing</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="contributing.html">Contributing to MDAnalysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="contributing_code.html">Contributing to the main codebase</a></li>
<li class="toctree-l1"><a class="reference internal" href="contributing_docs.html">Contributing to the user guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="preparing_releases_and_hotfixes.html">Preparing a release</a></li>
<li class="toctree-l1"><a class="reference internal" href="module_imports.html">Module imports in MDAnalysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="testing.html">Tests in MDAnalysis</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">References</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#citations-using-duecredit">Citations using Duecredit</a></li>
</ul>
</li>
</ul>

</div>

      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">MDAnalysis User Guide</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">References</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/references.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="references">
<span id="id1"></span><h1>References<a class="headerlink" href="#references" title="Permalink to this heading"></a></h1>
<p>MDAnalysis and the included algorithms are scientific software that
are described in academic publications. <strong>Please cite these papers when you use
MDAnalysis in published work.</strong></p>
<p>It is possible to <a class="reference external" href="https://docs.mdanalysis.org/2.7.0-dev0/documentation_pages/references.html#citations-using-duecredit" title="(in MDAnalysis v2.7.0-dev0)"><span class="xref std std-ref">automatically generate a list of references</span></a> for any program that uses
MDAnalysis. This list (in common reference manager formats) contains
the citations associated with the specific algorithms and libraries
that were used in the program.</p>
<section id="citations-using-duecredit">
<h2>Citations using Duecredit<a class="headerlink" href="#citations-using-duecredit" title="Permalink to this heading"></a></h2>
<p>Citations can be automatically generated using <a class="reference external" href="https://github.com/duecredit/duecredit">duecredit</a>, depending on the
packages used. Duecredit is easy to install via <code class="docutils literal notranslate"><span class="pre">pip</span></code>. Simply type:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pip<span class="w"> </span>install<span class="w"> </span>duecredit
</pre></div>
</div>
<p><a class="reference external" href="https://github.com/duecredit/duecredit">duecredit</a> will remain an optional dependency, i.e. any code using
MDAnalysis will work correctly even without duecredit installed.</p>
<p>A list of citations for <code class="docutils literal notranslate"><span class="pre">yourscript.py</span></code> can be obtained using simple
commands.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>/path/to/yourmodule
python<span class="w"> </span>-m<span class="w"> </span>duecredit<span class="w"> </span>yourscript.py
</pre></div>
</div>
<p>or set the environment variable <span class="target" id="index-0"></span><code class="xref std std-envvar docutils literal notranslate"><span class="pre">DUECREDIT_ENABLE</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>DUECREDIT-ENABLE<span class="o">=</span>yes<span class="w"> </span>python<span class="w"> </span>yourscript.py
</pre></div>
</div>
<p>Once the citations have been extracted (to a hidden file in the
current directory), you can use the <strong class="program">duecredit</strong> program to
export them to different formats. For example, one can display them in
BibTeX format, using:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>duecredit<span class="w"> </span>summary<span class="w"> </span>--format<span class="o">=</span>bibtex
</pre></div>
</div>
<p><strong>Please cite your use of MDAnalysis and the packages and algorithms
that it uses. Thanks!</strong></p>
<div class="docutils container" id="id2">
<div class="citation" id="id32" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>ALB93<span class="fn-bracket">]</span></span>
<p>Andrea Amadei, Antonius B. M. Linssen, and Herman J. C. Berendsen. Essential dynamics of proteins. <em>Proteins: Structure, Function, and Bioinformatics</em>, 17(4):412–425, 1993. _eprint: https://onlinelibrary.wiley.com/doi/pdf/10.1002/prot.340170408. URL: <a class="reference external" href="http://onlinelibrary.wiley.com/doi/abs/10.1002/prot.340170408">http://onlinelibrary.wiley.com/doi/abs/10.1002/prot.340170408</a> (visited on 2021-01-05), <a class="reference external" href="https://doi.org/https://doi.org/10.1002/prot.340170408">doi:https://doi.org/10.1002/prot.340170408</a>.</p>
</div>
<div class="citation" id="id4" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>BKV00<span class="fn-bracket">]</span></span>
<p>M. Bansal, S. Kumar, and R. Velavan. HELANAL: a program to characterize helix geometry in proteins. <em>Journal of Biomolecular Structure &amp; Dynamics</em>, 17(5):811–819, April 2000. 00175. <a class="reference external" href="https://doi.org/10.1080/07391102.2000.10506570">doi:10.1080/07391102.2000.10506570</a>.</p>
</div>
<div class="citation" id="id14" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>BDPW09<span class="fn-bracket">]</span></span>
<p>Oliver Beckstein, Elizabeth J. Denning, Juan R. Perilla, and Thomas B. Woolf. Zipping and Unzipping of Adenylate Kinase: Atomistic Insights into the Ensemble of Open↔Closed Transitions. <em>Journal of Molecular Biology</em>, 394(1):160–176, November 2009. 00107. URL: <a class="reference external" href="https://linkinghub.elsevier.com/retrieve/pii/S0022283609011164">https://linkinghub.elsevier.com/retrieve/pii/S0022283609011164</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.1016/j.jmb.2009.09.009">doi:10.1016/j.jmb.2009.09.009</a>.</p>
</div>
<div class="citation" id="id16" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>BHE13<span class="fn-bracket">]</span></span>
<p>R. B. Best, G. Hummer, and W. A. Eaton. Native contacts determine protein folding mechanisms in atomistic simulations. <em>Proceedings of the National Academy of Sciences</em>, 110(44):17874–17879, October 2013. 00259. URL: <a class="reference external" href="http://www.pnas.org/cgi/doi/10.1073/pnas.1311599110">http://www.pnas.org/cgi/doi/10.1073/pnas.1311599110</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.1073/pnas.1311599110">doi:10.1073/pnas.1311599110</a>.</p>
</div>
<div class="citation" id="id24" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>CL06<span class="fn-bracket">]</span></span>
<p>Ronald R. Coifman and Stéphane Lafon. Diffusion maps. <em>Applied and Computational Harmonic Analysis</em>, 21(1):5–30, July 2006. 02271. <a class="reference external" href="https://doi.org/10.1016/j.acha.2006.04.006">doi:10.1016/j.acha.2006.04.006</a>.</p>
</div>
<div class="citation" id="id25" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>dlPHHvdW08<span class="fn-bracket">]</span></span>
<p>J. de la Porte, B. M. Herbst, W. Hereman, and S. J. van der Walt. An introduction to diffusion maps. In <em>In The 19th Symposium of the Pattern Recognition Association of South Africa</em>. 2008. 00038.</p>
</div>
<div class="citation" id="id27" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>FPKD11<span class="fn-bracket">]</span></span>
<p>Andrew Ferguson, Athanassios Z. Panagiotopoulos, Ioannis G. Kevrekidis, and Pablo G. Debenedetti. Nonlinear dimensionality reduction in molecular simulation: The diffusion map approach. <em>Chemical Physics Letters</em>, 509(1-3):1–11, June 2011. 00085. <a class="reference external" href="https://doi.org/10.1016/j.cplett.2011.04.066">doi:10.1016/j.cplett.2011.04.066</a>.</p>
</div>
<div class="citation" id="id17" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>FKDD07<span class="fn-bracket">]</span></span>
<p>Joel Franklin, Patrice Koehl, Sebastian Doniach, and Marc Delarue. MinActionPath: maximum likelihood trajectory for large-scale structural transitions in a coarse-grained locally harmonic energy landscape. <em>Nucleic Acids Research</em>, 35(suppl_2):W477–W482, July 2007. 00083. URL: <a class="reference external" href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkm342">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkm342</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.1093/nar/gkm342">doi:10.1093/nar/gkm342</a>.</p>
</div>
<div class="citation" id="id3" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>GLB+16<span class="fn-bracket">]</span></span>
<p>Richard J. Gowers, Max Linke, Jonathan Barnoud, Tyler J. E. Reddy, Manuel N. Melo, Sean L. Seyler, Jan Domański, David L. Dotson, Sébastien Buchoux, Ian M. Kenney, and Oliver Beckstein. MDAnalysis: A Python Package for the Rapid Analysis of Molecular Dynamics Simulations. <em>Proceedings of the 15th Python in Science Conference</em>, pages 98–105, 2016. 00152. URL: <a class="reference external" href="https://conference.scipy.org/proceedings/scipy2016/oliver_beckstein.html">https://conference.scipy.org/proceedings/scipy2016/oliver_beckstein.html</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.25080/Majora-629e541a-00e">doi:10.25080/Majora-629e541a-00e</a>.</p>
</div>
<div class="citation" id="id5" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>HKP+07<span class="fn-bracket">]</span></span>
<p>Benjamin A. Hall, Samantha L. Kaye, Andy Pang, Rafael Perera, and Philip C. Biggin. Characterization of Protein Conformational States by Normal-Mode Frequencies. <em>Journal of the American Chemical Society</em>, 129(37):11394–11401, September 2007. 00020. URL: <a class="reference external" href="https://doi.org/10.1021/ja071797y">https://doi.org/10.1021/ja071797y</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.1021/ja071797y">doi:10.1021/ja071797y</a>.</p>
</div>
<div class="citation" id="id30" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>Hes02<span class="fn-bracket">]</span></span>
<p>Berk Hess. Convergence of sampling in protein simulations. <em>Physical Review E</em>, 65(3):031910, March 2002. 00348. URL: <a class="reference external" href="https://link.aps.org/doi/10.1103/PhysRevE.65.031910">https://link.aps.org/doi/10.1103/PhysRevE.65.031910</a> (visited on 2020-03-07), <a class="reference external" href="https://doi.org/10.1103/PhysRevE.65.031910">doi:10.1103/PhysRevE.65.031910</a>.</p>
</div>
<div class="citation" id="id18" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>JWLM78<span class="fn-bracket">]</span></span>
<p>Joël Janin, Shoshanna Wodak, Michael Levitt, and Bernard Maigret. Conformation of amino acid side-chains in proteins. <em>Journal of Molecular Biology</em>, 125(3):357 – 386, 1978. 00874. URL: <a class="reference external" href="http://www.sciencedirect.com/science/article/pii/0022283678904084">http://www.sciencedirect.com/science/article/pii/0022283678904084</a>, <a class="reference external" href="https://doi.org/10.1016/0022-2836(78)90408-4">doi:10.1016/0022-2836(78)90408-4</a>.</p>
</div>
<div class="citation" id="id33" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>Jol02<span class="fn-bracket">]</span></span>
<p>I. T. Jolliffe. <em>Principal Component Analysis</em>. Springer Series in Statistics. Springer-Verlag, New York, 2 edition, 2002. ISBN 978-0-387-95442-4. URL: <a class="reference external" href="http://www.springer.com/gp/book/9780387954424">http://www.springer.com/gp/book/9780387954424</a> (visited on 2021-01-05), <a class="reference external" href="https://doi.org/10.1007/b98835">doi:10.1007/b98835</a>.</p>
</div>
<div class="citation" id="id12" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>LAT09<span class="fn-bracket">]</span></span>
<p>Pu Liu, Dimitris K. Agrafiotis, and Douglas L. Theobald. Fast determination of the optimal rotational matrix for macromolecular superpositions. <em>Journal of Computational Chemistry</em>, pages n/a–n/a, 2009. URL: <a class="reference external" href="http://doi.wiley.com/10.1002/jcc.21439">http://doi.wiley.com/10.1002/jcc.21439</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.1002/jcc.21439">doi:10.1002/jcc.21439</a>.</p>
</div>
<div class="citation" id="id19" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>LDA+03<span class="fn-bracket">]</span></span>
<p>Simon C. Lovell, Ian W. Davis, W. Bryan Arendall, Paul I. W. de Bakker, J. Michael Word, Michael G. Prisant, Jane S. Richardson, and David C. Richardson. Structure validation by Cα geometry: ϕ,ψ and Cβ deviation. <em>Proteins: Structure, Function, and Bioinformatics</em>, 50(3):437–450, January 2003. 03997. URL: <a class="reference external" href="http://doi.wiley.com/10.1002/prot.10286">http://doi.wiley.com/10.1002/prot.10286</a> (visited on 2020-02-06), <a class="reference external" href="https://doi.org/10.1002/prot.10286">doi:10.1002/prot.10286</a>.</p>
</div>
<div class="citation" id="id31" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>MLS09<span class="fn-bracket">]</span></span>
<p>Gia G. Maisuradze, Adam Liwo, and Harold A. Scheraga. Principal component analysis for protein folding dynamics. <em>Journal of molecular biology</em>, 385(1):312–329, January 2009. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2652707/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2652707/</a> (visited on 2020-12-27), <a class="reference external" href="https://doi.org/10.1016/j.jmb.2008.10.018">doi:10.1016/j.jmb.2008.10.018</a>.</p>
</div>
<div class="citation" id="id11" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>MADWB11<span class="fn-bracket">]</span></span>
<p>Naveen Michaud-Agrawal, Elizabeth J. Denning, Thomas B. Woolf, and Oliver Beckstein. MDAnalysis: A toolkit for the analysis of molecular dynamics simulations. <em>Journal of Computational Chemistry</em>, 32(10):2319–2327, July 2011. 00778. URL: <a class="reference external" href="http://doi.wiley.com/10.1002/jcc.21787">http://doi.wiley.com/10.1002/jcc.21787</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.1002/jcc.21787">doi:10.1002/jcc.21787</a>.</p>
</div>
<div class="citation" id="id15" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>NCR18<span class="fn-bracket">]</span></span>
<p>Hai Nguyen, David A Case, and Alexander S Rose. NGLview–interactive molecular graphics for Jupyter notebooks. <em>Bioinformatics</em>, 34(7):1241–1242, April 2018. 00024. URL: <a class="reference external" href="https://academic.oup.com/bioinformatics/article/34/7/1241/4721781">https://academic.oup.com/bioinformatics/article/34/7/1241/4721781</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btx789">doi:10.1093/bioinformatics/btx789</a>.</p>
</div>
<div class="citation" id="id26" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>RZMC11<span class="fn-bracket">]</span></span>
<p>Mary A. Rohrdanz, Wenwei Zheng, Mauro Maggioni, and Cecilia Clementi. Determination of reaction coordinates via locally scaled diffusion map. <em>The Journal of Chemical Physics</em>, 134(12):124116, March 2011. 00220. <a class="reference external" href="https://doi.org/10.1063/1.3569857">doi:10.1063/1.3569857</a>.</p>
</div>
<div class="citation" id="id23" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SB17<span class="fn-bracket">]</span></span>
<p>Sean Seyler and Oliver Beckstein. Molecular dynamics trajectory for benchmarking MDAnalysis. June 2017. 00002. URL: <a class="reference external" href="https://figshare.com/articles/Molecular_dynamics_trajectory_for_benchmarking_MDAnalysis/5108170">https://figshare.com/articles/Molecular_dynamics_trajectory_for_benchmarking_MDAnalysis/5108170</a> (visited on 2020-02-09), <a class="reference external" href="https://doi.org/10.6084/m9.figshare.5108170.v1">doi:10.6084/m9.figshare.5108170.v1</a>.</p>
</div>
<div class="citation" id="id13" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SKTB15<span class="fn-bracket">]</span></span>
<p>Sean L. Seyler, Avishek Kumar, M. F. Thorpe, and Oliver Beckstein. Path Similarity Analysis: A Method for Quantifying Macromolecular Pathways. <em>PLOS Computational Biology</em>, 11(10):e1004568, October 2015. URL: <a class="reference external" href="https://dx.plos.org/10.1371/journal.pcbi.1004568">https://dx.plos.org/10.1371/journal.pcbi.1004568</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.1371/journal.pcbi.1004568">doi:10.1371/journal.pcbi.1004568</a>.</p>
</div>
<div class="citation" id="id29" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SFPG+19<span class="fn-bracket">]</span></span>
<p>Max Linke Shujie Fan, Ioannis Paraskevakos, Richard J. Gowers, Michael Gecht, and Oliver Beckstein. PMDA - Parallel Molecular Dynamics Analysis. In Chris Calloway, David Lippa, Dillon Niederhut, and David Shupe, editors, <em>Proceedings of the 18th Python in Science Conference</em>, 134 – 142. 2019. <a class="reference external" href="https://doi.org/10.25080/Majora-7ddc1dd1-013">doi:10.25080/Majora-7ddc1dd1-013</a>.</p>
</div>
<div class="citation" id="id34" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SJS14<span class="fn-bracket">]</span></span>
<p>Florian Sittel, Abhinav Jain, and Gerhard Stock. Principal component analysis of molecular dynamics: on the use of Cartesian vs. internal coordinates. <em>The Journal of Chemical Physics</em>, 141(1):014111, July 2014. <a class="reference external" href="https://doi.org/10.1063/1.4885338">doi:10.1063/1.4885338</a>.</p>
</div>
<div class="citation" id="id35" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SS18<span class="fn-bracket">]</span></span>
<p>Florian Sittel and Gerhard Stock. Perspective: Identification of collective variables and metastable states of protein dynamics. <em>The Journal of Chemical Physics</em>, 149(15):150901, October 2018. Publisher: American Institute of Physics. URL: <a class="reference external" href="http://aip.scitation.org/doi/10.1063/1.5049637">http://aip.scitation.org/doi/10.1063/1.5049637</a> (visited on 2021-01-05), <a class="reference external" href="https://doi.org/10.1063/1.5049637">doi:10.1063/1.5049637</a>.</p>
</div>
<div class="citation" id="id21" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SGW93<span class="fn-bracket">]</span></span>
<p>O S Smart, J M Goodfellow, and B A Wallace. The pore dimensions of gramicidin A. <em>Biophysical Journal</em>, 65(6):2455–2460, December 1993. 00522. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1225986/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1225986/</a> (visited on 2020-02-10), <a class="reference external" href="https://doi.org/10.1016/S0006-3495(93)81293-1">doi:10.1016/S0006-3495(93)81293-1</a>.</p>
</div>
<div class="citation" id="id20" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SNW+96<span class="fn-bracket">]</span></span>
<p>O. S. Smart, J. G. Neduvelil, X. Wang, B. A. Wallace, and M. S. Sansom. HOLE: a program for the analysis of the pore dimensions of ion channel structural models. <em>Journal of Molecular Graphics</em>, 14(6):354–360, 376, December 1996. 00935. <a class="reference external" href="https://doi.org/10.1016/s0263-7855(97)00009-x">doi:10.1016/s0263-7855(97)00009-x</a>.</p>
</div>
<div class="citation" id="id22" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SFSB14<span class="fn-bracket">]</span></span>
<p>Lukas S. Stelzl, Philip W. Fowler, Mark S. P. Sansom, and Oliver Beckstein. Flexible gates generate occluded intermediates in the transport cycle of LacY. <em>Journal of Molecular Biology</em>, 426(3):735–751, February 2014. 00000. URL: <a class="reference external" href="https://asu.pure.elsevier.com/en/publications/flexible-gates-generate-occluded-intermediates-in-the-transport-c">https://asu.pure.elsevier.com/en/publications/flexible-gates-generate-occluded-intermediates-in-the-transport-c</a> (visited on 2020-02-10), <a class="reference external" href="https://doi.org/10.1016/j.jmb.2013.10.024">doi:10.1016/j.jmb.2013.10.024</a>.</p>
</div>
<div class="citation" id="id36" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SM67<span class="fn-bracket">]</span></span>
<p>Hiromu Sugeta and Tatsuo Miyazawa. General method for calculating helical parameters of polymer chains from bond lengths, bond angles, and internal-rotation angles. <em>Biopolymers</em>, 5(7):673–679, 1967. URL: <a class="reference external" href="https://onlinelibrary.wiley.com/doi/abs/10.1002/bip.1967.360050708">https://onlinelibrary.wiley.com/doi/abs/10.1002/bip.1967.360050708</a>, <a class="reference external" href="https://arxiv.org/abs/https://onlinelibrary.wiley.com/doi/pdf/10.1002/bip.1967.360050708">arXiv:https://onlinelibrary.wiley.com/doi/pdf/10.1002/bip.1967.360050708</a>, <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bip.1967.360050708">doi:https://doi.org/10.1002/bip.1967.360050708</a>.</p>
</div>
<div class="citation" id="id10" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>The05<span class="fn-bracket">]</span></span>
<p>Douglas L. Theobald. Rapid calculation of RMSDs using a quaternion-based characteristic polynomial. <em>Acta Crystallographica Section A Foundations of Crystallography</em>, 61(4):478–480, July 2005. 00127. URL: <a class="reference external" href="http://scripts.iucr.org/cgi-bin/paper?S0108767305015266">http://scripts.iucr.org/cgi-bin/paper?S0108767305015266</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.1107/S0108767305015266">doi:10.1107/S0108767305015266</a>.</p>
</div>
<div class="citation" id="id6" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>TPB+15<span class="fn-bracket">]</span></span>
<p>Matteo Tiberti, Elena Papaleo, Tone Bengtsen, Wouter Boomsma, and Kresten Lindorff-Larsen. ENCORE: Software for Quantitative Ensemble Comparison. <em>PLOS Computational Biology</em>, 11(10):e1004415, October 2015. 00031. URL: <a class="reference external" href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004415">https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004415</a> (visited on 2020-02-05), <a class="reference external" href="https://doi.org/10.1371/journal.pcbi.1004415">doi:10.1371/journal.pcbi.1004415</a>.</p>
</div>
<div class="citation" id="id28" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>Wel62<span class="fn-bracket">]</span></span>
<p>B. P. Welford. Note on a Method for Calculating Corrected Sums of Squares and Products. <em>Technometrics</em>, 4(3):419–420, August 1962. URL: <a class="reference external" href="https://amstat.tandfonline.com/doi/abs/10.1080/00401706.1962.10490022">https://amstat.tandfonline.com/doi/abs/10.1080/00401706.1962.10490022</a> (visited on 2020-02-10), <a class="reference external" href="https://doi.org/10.1080/00401706.1962.10490022">doi:10.1080/00401706.1962.10490022</a>.</p>
</div>
</div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="testing.html" class="btn btn-neutral float-left" title="Tests in MDAnalysis" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2019-2023, Lily Wang, Irfan Alibay, Patricio Barletta, Cédric Bouysset, Jan Domanski, Bjarne Feddersen, Florent Langenfeld, Hugo MacDermott-Opeskin, Micaela Matta, Rocco Meli, Paul Smith, Mieczyslaw Torchala, Mark Verma, Yuxuan Zhuang, Richard J. Gowers, and Oliver Beckstein..</p>
  </div>

  

<div class="footer"><p>Please see
    our <a href="https://www.mdanalysis.org/pages/privacy/">Privacy Policy</a>
    to learn how <a href="https://www.mdanalysis.org">MDAnalysis</a> collects data.</p>
    <script data-goatcounter="https://mdanalysis.goatcounter.com/count"
        async src="//gc.zgo.at/count.js"></script>
</div>



</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
    var versions_json_url = 'https://userguide.mdanalysis.org/versions.json'
</script>

<div class="rst-versions" data-toggle="rst-versions" role="note"
     aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      <span class="fa fa-book"></span>
        
      <span class="fa fa-caret-down"></span>
    </span>

    <div class="rst-other-versions">
        <dl id="versionselector">
            <dt>Other Versions</dt>
        </dl>

    </div>
</div><script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>